SensusService/SensusServiceHelper.cs

```diff
@@ -1645,23 +1573,67 @@ public void AssertNotOnMainThread(string actionDescription)
 
         public void StopProtocols()
         {
-            lock (_registeredProtocols)
-            {
-                _logger.Log("Stopping protocols.", LoggingLevel.Normal, GetType());
+            _logger.Log("Stopping protocols.", LoggingLevel.Normal, GetType());
 
-                foreach (Protocol protocol in _registeredProtocols)
-                    if (protocol.Running)
-                    {
-                        try
-                        {
-                            protocol.Stop();
-                        }
-                        catch (Exception ex)
-                        {
-                            _logger.Log("Failed to stop protocol \"" + protocol.Name + "\":  " + ex.Message, LoggingLevel.Normal, GetType());
-                        }
-                    }
+            foreach (var protocol in _registeredProtocols.ToArray().Where(p => p.Running))
+            {
+                try
+                {
+                    protocol.Stop();
+                }
+                catch (Exception ex)
+                {
+                    _logger.Log($"Failed to stop protocol \"{protocol.Name}\": {ex.Message}", LoggingLevel.Normal, GetType());
+                }
             }
-        }
+        }
+
+        #region Private Methods
+
+        private void RemoveScripts(bool issueNotification, params Script[] scripts)
+        {
+            var removed = false;
+
+            foreach (var script in scripts)
+            {
+                removed = _scriptsToRun.Remove(script);

```

Comment:
Won't be correct when multiple scripts are passed in, one is removed, and a later one is not.